azure/image/Dockerfile

```diff
@@ -11,84 +11,23 @@ ARG BINARIES_IMG="quay.io/confidential-containers/podvm-binaries-ubuntu-amd64"
 FROM ${BINARIES_IMG} AS podvm_binaries
 FROM ${BUILDER_IMG} AS podvm_builder
 
-ARG CLOUD_PROVIDER=azure
 ARG PODVM_DISTRO=ubuntu
+ENV PODVM_DISTRO ${PODVM_DISTRO}
+
 # If not provided, uses system architecture
 ARG ARCH=x86_64
-ARG CAA_SRC=""
-ARG CAA_SRC_REF=""
-
-ENV CLOUD_PROVIDER ${CLOUD_PROVIDER}
-ENV PODVM_DISTRO ${PODVM_DISTRO}
 
-RUN if [ -n "${CAA_SRC}" ]; then \
-      rm -rf cloud-api-adaptor && \
-      git clone ${CAA_SRC} cloud-api-adaptor;\
-    fi && \
-    if [ -n "${CAA_SRC_REF}" ]; then \
-      cd cloud-api-adaptor && \
-      git fetch origin ${CAA_SRC_REF} && \
-      git checkout FETCH_HEAD -b ${CAA_SRC_REF} ;\
-    fi
+RUN rm -rf /src/cloud-api-adaptor
+COPY . /src/cloud-api-adaptor
```

Comment:
sorry, i missed that, what is copied here?

Reply:
Added a comment that it is copying the local source, overriding the code included in it.

Reply:
In general, it is preferable to minimize the files to be copied as build context.

In the case of the `cloud-api-adaptor` container image, the files that are actually necessary to build the` cloud-api-adaptor` are selectively copied. This style reduces time to copy files, and also is friendly to docker build cache.

https://github.com/confidential-containers/cloud-api-adaptor/blob/ade303e0acddcae9aa94bc753866578215587ae7/Dockerfile#L14-L20

But, selectively copying files will break the logic of pod VM image name generation described in my above comment. So, I think it is OK for now, and we can optimize this in a later PR.

Reply:
Imho we should not COPY or ADD files from the local host to ensure isolated builds are possible. This was the reasoning behind not having it. The existing Dockerfiles does have logic to use custom fork/branch and that should suffice for the custom workflows. Alternatively we can use volume mount during build if required. But I don't think this functionality is available with docker/buildx yet. It's available in podman/buildah

Reply:
isn't `COPY . /src/cloud-api-adaptor` copies current dir (azure/image) as source?